whisper/src/bin/whisper-fetch.rs

```diff
@@ -87,33 +84,31 @@ fn main() -> Result<(), Error> {
         Some(ref s) => return Err(format_err!("No such drop option {}.", s)),
     };
 
-    let values: Vec<String> = archive
-        .values
-        .into_iter()
-        .filter(&filter)
-        .map(|v| v.map(|x| x.to_string()).unwrap_or("None".to_owned()))
-        .collect();
+    let archive = file.fetch(seconds_per_point, interval, now)?
+        .ok_or(err_msg("No data in selected timerange"))?
+        .filter_out(&filter);
```

Comment:
filter_map?
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map

Reply:
For what purposes? Filter is inside the structure, like @andy128k proposed to do in previous PR review, so filter_out returns the same struct ArchiveData.